app/src/main/java/io/homeassistant/companion/android/notifications/MessagingManager.kt

```diff
@@ -1233,6 +1245,42 @@ class MessagingManager @Inject constructor(
             return@withContext image
         }
 
+    @RequiresApi(Build.VERSION_CODES.UPSIDE_DOWN_CAKE)
+    private suspend fun saveTempAnimatedImage(serverId: Int, url: URL?, requiresAuth: Boolean = false): Uri? =
+        withContext(
+            Dispatchers.IO
+        ) {
+            if (url == null || url.path.endsWith("gif").not()) {
+                return@withContext null
+            }
+
+            // delete previous images that are no longer needed
+            val imageCutoff = LocalDateTime.now().minusHours(1)
+            context.externalCacheDir?.listFiles()?.filter { file ->
+                file.endsWith("_animated_notification.gif") &&
```

Comment:
This always returns `false`, probably because of the confusing behavior where it tries comparing components? Suggestion: `file.path.endsWith`.